chore(multi-region): add operational simplification from Zeebe 8.6.

Langleu opened the pull request:
## Description

<!-- Provide an overview of what to expect in the PR. -->
<!-- Link to an associated epic when applicable. -->
<!-- Add an assignee and use `component:` or version labels for good hygiene! -->

related to https://github.com/camunda/team-infrastructure-experience/issues/325 and https://github.com/camunda/product-hub/issues/2173.

This ticket contains the simplification of the dual-region operational procedure based on the new 8.6 changes in Zeebe.

It reworks for next (8.6):
- the pictures of the operational procedure
- the steps that need to be taken
- the concept page including RPO / RTO
- some smaller fixes here and there

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
    - either alpha5 or latest stable would be nice.
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

Langleu commented:
The change is intentional since 8.5 still uses the old procedure while was reworked for next (8.6)

Langleu commented:
Great to hear @conceptualshark 🙇 🚀!
Since it's a change of an existing document, I think it would be okay to have your review already!
The only thing I could imagine is that maybe a step has to be moved, but the content itself will remain.

deepthidevaki commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
````diff
@@ -605,7 +608,7 @@ kubectl --context $CLUSTER_SURVIVING get deployments $HELM_RELEASE_NAME-operate
 ```
 
-For the Zeebe Elasticsearch exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful disabling.
+For the Zeebe Elasticsearch exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful disabling. This is a synchronous operation.
````
You can also link the api docs above http://localhost:3000/docs/next/self-managed/zeebe-deployment/operations/management-api/#exporting-api

Langleu replied:
linked it where it's executed.

deepthidevaki commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
````diff
@@ -1000,38 +917,22 @@ desired={<Thirteen viewBox="140 40 680 500" />}
 ```
 
-3. Reactivate the exporters by sending the API activation request via the Zeebe Gateway:
+2. Reactivate the exporters by sending the API activation request via the Zeebe Gateway:
````
It would be better to split this into a new step. We should resume only after the exporter has enabled. Enabling exporter usually finishes faster, but it is not a synchronous operation. So the process should be as follows.

1. enable exporter
2. verify it is enabled
3. Resume exporting

deepthidevaki commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
````diff
@@ -1042,121 +943,179 @@ curl -i localhost:9600/actuator/exporting/resume -XPOST
-kubectl --context $CLUSTER_SURVIVING port-forward services/$HELM_RELEASE_NAME-zeebe-gateway 26500:26500 -n $CAMUNDA_NAMESPACE_SURVIVING
-zbctl status --insecure --address localhost:26500
-```
+For the Zeebe Elasticsearch exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful resumption. This is a synchronous operation.
````
```suggestion
For the reactivating the exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful resumption. This is a synchronous operation.
```

deepthidevaki commented on docs/self-managed/concepts/multi-region/dual-region.md:
```diff
@@ -196,12 +190,12 @@ The **Recovery Time Objective (RTO)** is the time to restore services to a funct
 For Zeebe the **RPO** is **0**.
 
-For Operate and Tasklist the **RPO** is close to **0** for critical data due to the previously mentioned small chance of data loss in Elasticsearch during the failback procedure.
+For Operate and Tasklist the **RPO** is **0**.
```
Can be combined with the previous sentence.

deepthidevaki commented on docs/self-managed/concepts/multi-region/dual-region.md:
```diff
@@ -196,12 +190,12 @@ The **Recovery Time Objective (RTO)** is the time to restore services to a funct
-- **failover** has an **RTO** of **15-20** minutes to restore a functional state, excluding DNS considerations.
-- **failback** has an **RTO** of **25-30 + X** minutes to restore a functional state. Where X is the time it takes to back up and restore Elasticsearch, which is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
+- **failover** has an **RTO** of **< 1** minute to restore a functional state, excluding DNS considerations.
+- **failback** has an **RTO** of **5 + X** minutes to restore a functional state. Where X is the time it takes to back up and restore Elasticsearch, which is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
```
:question: What is the reason for 5m?

Langleu replied:
I took the estimate from our automated tests for the procedure and comes mostly from installing C8.
First install of C8 and waiting for Elastic to be ready is roughly 2-3 minutes.
Later installing Operate / Tasklist is another minute.
In-between are the steps that need to be taken when automated of roughly a minute. (It's automated, so a manual execution will even take longer).

Langleu replied:
Since RTO contains the time to recover as well, it's roughly 5 minutes from my side.
I'm open to other suggestions if you think it should be more or less.

deepthidevaki replied:
Thanks. I'm wondering if we have to specify the 5 minutes. Can we say it in general that it depends on the ES backup and restore + some additional time that needs to install and start C8 components in the new region? 5 minutes seems arbitrary and can also depends on whether it is manual, automated etc.

Langleu replied:
I get your point about the arbitrary number, but we had before 25-30 minutes.
Just saying it takes whatever time for Backup and whatever time for Install sounds very fuzzy and questionable on why it can't be pinpoint.
I mean Elastic + WebApps have health checks in place, that's why there is a guaranteed lower bound anyway.
Sure there are more factors that can play in but I feel like giving 0 guidance is even less useful.

How about we rephase it to:

> - **failback** has an **RTO** of **5 + X** minutes to restore a functional state. Where X is the time it takes to back up and restore Elasticsearch, which is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
During our automated tests, the reinstallation and reconfiguration of Camunda 8 takes 5 minutes. This can serve as a general guideline for the time required, though your experience may vary depending on your available resources and familiarity with the operational procedure.

deepthidevaki replied:
The rephrase sounds good to me. :+1:

deepthidevaki reviewed (dismissed):
:rocket:  

One minor thing to change:

In Failover Step 1, it is not necessary to ensure network isolation between two regions anymore. Previously, we had to do it because we were recreating brokers with the same id on the other region which could create conflicts if the network was not isolated and both the old and new brokers exists at the same time.

conceptualshark commented on docs/self-managed/concepts/multi-region/dual-region.md:
```diff
@@ -129,11 +127,7 @@ In the event of a total active region loss, the following data will be lost:
-- During the failback procedure, there’s a small chance that some data will be lost in Elasticsearch affecting Operate and Tasklist.
-  - This **does not** affect the processing of process instances in any way. The impact is that some information about the affected instances might not be visible in Operate and Tasklist.
-  - This is further explained in the [operational procedure](./../../operational-guides/multi-region/dual-region-ops.md?failback=step2#failback) during the relevant step.
+- Connectors can be deployed alongside but ensure to understand idempotency based on [the described documentation](../../../components/connectors/use-connectors/inbound.md#creating-the-connector-event).
```
Could this be more declarative regarding support?

```suggestion
- Connectors are supported, but may require special considerations regarding [idempotency](../../../components/connectors/use-connectors/inbound.md#creating-the-connector-event).
```
It's not clear to me from the linked documentation why idempotency would be more of a concern in this setup, but that may be my lack of familiarity. Is it _always_ a concern in this setup? Should they always use the message start event element? Some clarity here would be great.

Langleu replied:
I don't want to use the word "supported" atm since we have an extra ticket to actually create tests / manifest adjustments etc. to officially support it.
It's just that it isn't a limitation anymore but can be done under premise to understand message idempotency.

I've added a subsentence on why it's important.
TL;DR - multiple connectors instances + message idempotency = no duplicated events.
Without it = may end up triggering things multiple times.
Same applies to a single region setup, just that in multi-region by default we have two connector instances.

conceptualshark replied:
This works for me then. 👍 The additional messaging helps!

conceptualshark commented on docs/self-managed/concepts/multi-region/dual-region.md:
```diff
@@ -196,12 +190,12 @@ The **Recovery Time Objective (RTO)** is the time to restore services to a funct
-- **failover** has an **RTO** of **15-20** minutes to restore a functional state, excluding DNS considerations.
-- **failback** has an **RTO** of **25-30 + X** minutes to restore a functional state. Where X is the time it takes to back up and restore Elasticsearch, which is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
+- **failover** has an **RTO** of **< 1** minute to restore a functional state, excluding DNS considerations.
+- **failback** has an **RTO** of **5 + X** minutes to restore a functional state. Where X is the time it takes to back up and restore Elasticsearch, which is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
```
```suggestion
- **failback** has an **RTO** of **5 + X** minutes to restore a functional state, where X is the time it takes to back up and restore Elasticsearch. This timing is highly dependent on the setup and chosen [Elasticsearch backup type](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types).
```

conceptualshark commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
```diff
@@ -44,12 +44,20 @@ Running dual-region setups requires the users to be able to detect any regional
 
 :::
 
+:::info
+
+The operational procedure was reworked in Camunda 8.6 to be more straightforward and reliable.
+
+The old procedure of 8.5 [as outlined](../../../../versioned_docs/version-8.5/self-managed/operational-guides/multi-region/dual-region-ops.md) is still compatible with Camunda 8.6 but compatibility will be removed in 8.7.
+
+:::
+
```
```suggestion
:::info
This procedure has been updated in the Camunda 8.6 release. The procedure used in Camunda 8.5 has been deprecated, and compatibility will be removed in the 8.7 release.
:::
```
I'm so-so on if you'd like to include the link to the old version. If we're deprecating that method, I don't necessarily see the value in it, unless there are circumstances where the older procedure would be required. 

Can we move this up to the top of the page as well, for visibility?

Langleu replied:
I'm fine not including the link, a person can just use the navbar on top to quickly switch the version.
It was just for people to have a grace period to readjust their own documentation.

Sure moved it to the top.

conceptualshark commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
```diff
@@ -177,82 +183,82 @@ Due to the Zeebe data replication, no data has been lost.
 
-The chosen `camunda-values-failover.yml` requires adjustments before installing the Helm chart and the same has to be done for the base `camunda-values.yml` in `aws/dual-region/kubernetes`.
+- if it's exposed to the outside, one can skip port-forwarding and use the URL directly
+- one can use, for example, the Elasticsearch pod to exec into it and curl from there
```
```suggestion
- one can `exec` into an existing pod (such as Elasticsearch), and `curl` from there
```

^ Is this a correct rephrasing? Some clarity + command backtics.

Langleu replied:
sure, I also linked to the official k8s exec docs on the command.

conceptualshark commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
```diff
@@ -177,82 +183,82 @@ Due to the Zeebe data replication, no data has been lost.
-The following command will show the deployed pods of the failover namespace.
-
-Only the minimal amount of brokers required to restore the quorum will be deployed in the failover installation. For example, if `clusterSize` is eight, two Zeebe brokers will be deployed in the failover installation instead of the normal four. This is expected.
+3. Based on the [Cluster Scaling APIs](../../zeebe-deployment/operations/cluster-scaling.md) you will send a request to the Zeebe Gateway to redistribute the load to the remaining brokers, thereby removing the lost brokers.
```
```suggestion
3. Based on the [Cluster Scaling APIs](../../zeebe-deployment/operations/cluster-scaling.md), send a request to the Zeebe Gateway to redistribute the load to the remaining brokers, thereby removing the lost brokers.
```

conceptualshark reviewed (changes requested):
A few questions/suggestions, but otherwise this upgrade looks great!

Langleu commented on docs/self-managed/operational-guides/multi-region/dual-region-ops.md:
```diff
@@ -328,127 +345,83 @@ desired={<Seven viewBox="140 40 680 500" />}
-  --namespace $CAMUNDA_NAMESPACE_SURVIVING \
-  -f camunda-values.yml \
-  -f $REGION_SURVIVING/camunda-values.yml
+curl -XGET 'http://localhost:9600/actuator/exporters'
```
great! I left a todo in the text where we should probably link it and only have a wrong placeholder atm.

Langleu commented:
Thanks @deepthidevaki and @conceptualshark for the review 🙇.

@deepthidevaki:
- Good note on the isolation, I've added that information to the 8.5 procedure and removed it from 8.6

Besides that addressed all points except for the two that are left open, which we may have to realign on.

Langleu commented on versioned_docs/version-8.5/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md:
````diff
@@ -226,10 +226,10 @@ kubectl --context $CLUSTER_0 apply -f https://raw.githubusercontent.com/camunda/
 
 ```shell
-./generate_core_dns_entry.sh
+./generate_core_dns_entry.sh legacy
````
I'll be able to remove it in the versioning rework again but atm for completeness would add it.

Langleu commented on docs/self-managed/concepts/multi-region/dual-region.md:
```diff
@@ -54,8 +54,6 @@ The currently supported Camunda 8 Self-Managed components are:
 
```
okay, now I understand what you mean about having suggestions on readability.
I'd suggest doing this in a separate PR decoupled from this topic, so DevEx and others can properly review your suggestions.

Langleu commented on versioned_docs/version-8.5/self-managed/operational-guides/multi-region/dual-region-ops.md:
Same as above, would suggest doing that in a separate PR decoupled from this topic.

Langleu commented:
@conceptualshark could you do a last approval and feel free to merge afterwards!

I had to rebase the PR for the tests to pass.
Only changed since your approval:
- https://github.com/camunda/camunda-docs/pull/4205/commits/ad394f4e6135476b38b612c4190442b9478e45d8
- https://github.com/camunda/camunda-docs/pull/4205/commits/1ca09762fa978dd883f644ce820f89c8dc4fe559
